### PR TITLE
feat(RHINENG-12792): Migrate the Review (final) wizard step

### PR DESCRIPTION
**Pull request by gkarat**
Implements https://issues.redhat.com/browse/RHINENG-12792.

This decouples the FinishedCreatePolicy component into REST/GraphQL components depending on the feature flag set on/off respectively. `updatePolicy` for the REST component performs several steps to create a new policy: creates a policy using selected profile, assigns systems, fetches newly created tailorings and assigns rules to every tailoring.

## How to test

Go to the policies page and open a modal by clicking Create new policy. Choose any major OS, select a profile. Follow the systems and rules selections steps (these ones are not changed anyhow with this PR). In the final step, check that the overview contains correct OS selected, the number of systems and other details. Submit the creation and verify that the policy is successfully created. You should also get a link to the new policy page with the success notification.

## Screenshots
image

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by Fewwy** [comment]
Found 1 error that Georgy will fix, after that I will do another review

File: src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import React, { useCallback } from 'react';
+import FinishedCreatePolicyBase from './FinishedCreatePolicyBase';
+import useCreatePolicy from '../../../Utilities/hooks/api/useCreatePolicy';
+import useAssignRules from '../../../Utilities/hooks/api/useAssignRules';
+import useAssignSystems from '../../../Utilities/hooks/api/useAssignSystems';
+import useTailorings from '../../../Utilities/hooks/api/useTailorings';
+
+export const useUpdatePolicy = () => {
+  const { fetch: createPolicy } = useCreatePolicy({ skip: true });
```
Comment (mkholjuraev) [inline src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js:9]: Using a hook that has `ComplianceQuery` in the background that has a lot of unrelated stuff seems not right to me. Those unrelated things in `ComplianceQuery` are quite heavy in fact:
1. useSerialisedTableState (nothing from table logic is relevant to Post requests)
2. useQuery (this hooks has a lot that may build up complications)


 Instead, we could just use the plain function from js-client (I believe it has already axios interceptors).

Comment (gkarat) [reply src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js:9]: As I was stating earlier, we have to be consistent in API management. `unrelated stuff` what do you mean here? Using `skip` and setting `setDataState` to false limits the features to only those required for this POST request. Also, how do you imagine adding axios interceptors? I am not honestly willing to re-implement the tooling for every component.

File: src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicy.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import useAPIV2FeatureFlag from '../../../Utilities/hooks/useAPIV2FeatureFlag';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+import FinishedCreatePolicyRest from './FinishedCreatePolicyRest';
+import FinishedCreatePolicyGraphQL from './FinishedCreatePolicyGraphQL';
+
+const FinishedCreatePolicyWrapper = (props) => {
+  const apiV2Enabled = useAPIV2FeatureFlag();
+
+  return apiV2Enabled === undefined ? (
```
[the remark this answers is not in the document]
Comment (gkarat) [reply src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicy.js:10]: I am not sure it handles props passing properly. The thing is FinishedCreatePolicy can take some props and we make sure they are passed further to the chosen component. With GatedComponents, it's a bit less transparent and I am afraid props are discarded:

```
...
        <RestComponent />
      ) : (
        <GraphQLComponent />
```

File: src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js (hunk 1 above)
Comment (mkholjuraev) [reply src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js:9]: Regarding _unrelated stuff what do you mean here_: I believe I have provided already enough examples of `unrelated stuff` above. I have a feeling that we are increasing code complexity. I do understand the importance of  having the interceptors and I think there are easier ways to add it into api requests. In general, I am also sure that we can simplify much of the stuff we created in our migration.

Comment (mkholjuraev) [reply src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js:9]: What I am concerned about is that if we continue increasing complexity, we will end up again in a situation in which not everyone understands all the ins and outs of the codebase and we may face situations where devs avoid working with that complexity as it was in the past.

Comment (gkarat) [reply src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js:9]: I am not sure the axios interceptors question is answered. The useComplianceQuery hook is already established. I would refrain from discussing the architectural decisions in this PR.

**Review by Fewwy — APPROVED** [review]
Tested it with FF and without. Good job

**Comment by gkarat** [comment]
Some problem with node installation for ci.int. I will ignore this and merge if Konflux is green

File: src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js (hunk 1 above)
Comment (mkholjuraev) [reply src/SmartComponents/CreatePolicy/FinishedCreatePolicy/FinishedCreatePolicyRest.js:9]: Each PR can influence the complexity level. I think it is appropriate to discuss anything that may affect it :)

**Pull request merged**